coderd/insights.go

```diff
@@ -289,7 +293,7 @@ func (api *API) insightsTemplates(rw http.ResponseWriter, r *http.Request) {
 	// overhead from a transaction is not worth it.
 	eg.Go(func() error {
 		var err error
-		if interval != "" {
+		if interval != "" && slices.Contains(sections, codersdk.TemplateInsightsSectionIntervalReports) {
```

Comment:
Should we error if given interval without correct section?

Reply:
Added in `parseTemplateInsightsSections`